paper-menu.html

```diff
@@ -0,0 +1,330 @@
+        observer: 'activateEventChanged',
+        type: String,
+        value: 'click'
+      },
+
+      /**
+       * If this is set, only items with local name that matches the `selectable` are selectable.
+       *
+       * @attribute selectable
+       * @type String
+       */
+      selectable: {
+        type: String
+      }
+
+    },
+
+    hostAttributes: {
+      'role': 'menu',
+      'tabindex': '0'
+    },
+
+    listeners: {
+      'focus': 'onFocus',
+      'keydown': 'onKeydown'
+    },
+
+    created: function() {
+      this._bindActivateHandler = this.activateHandler.bind(this);
+    },
+
+    attached: function() {
+      this.selectableAttached();
+    },
+
+    detached: function() {
+      this.selectableDetached();
+      this.removeListener(this.activateEvent);
+    },
+
+    addListener: function(eventName) {
+      if (eventName) {
+        this.addEventListener(eventName, this._bindActivateHandler);
+      }
+    },
+
+    removeListener: function(eventName) {
+      if (eventName) {
+        this.removeEventListener(eventName, this._bindActivateHandler);
+      }
+    },
+
+    activateEventChanged: function(eventName, old) {
+      this.removeListener(old);
+      this.addListener(eventName);
+    },
+
+    focusedItemChanged: function(focusedItem, old) {
+      old && old.setAttribute('tabindex', '-1');
+      if (focusedItem) {
+        focusedItem.setAttribute('tabindex', '0');
+        focusedItem.focus();
+      }
+    },
+
+    multiChanged: function(multi) {
+      this.selection.multi = multi;
+      this.selectedChanged(this.selected);
+    },
+
+    selectedChanged: function(selected, old) {
+      this._selectedChanged(selected, old);
+    },
+
+    selectedItemChanged: function(selectedItem) {
+      this._setFocusedItem(Array.isArray(selectedItem) ? selectedItem[0] : selectedItem);
+    },
+
+    activateHandler: function(e) {
+      var t = e.target;
+      var items = this.items;
+      while (t && t != this) {
+        var i = items.indexOf(t);
+        if (i >= 0) {
+          if (t.hasAttribute('disabled')) {
+            return;
+          }
+          var value = this.indexToValue(i);
+          if (!this.fire('core-activate', {selected: value, item: t}).defaultPrevented) {
+            this.select(value);
+          }
+          return;
+        }
+        t = t.parentNode;
+      }
+    },
+
+    onFocus: function(event) {
+      // clear the cached focus item
+      this._setFocusedItem(null);
+      // focus the selected item when the menu receives focus, or the first item
+      // if no item is selected
+      var selectedItem = this.selectedItem;
+      selectedItem = Array.isArray(selectedItem) ? selectedItem[0] : selectedItem;
+      if (selectedItem) {
+        this._setFocusedItem(selectedItem);
+      } else {
+        this._setFocusedItem(this.items[0]);
+      }
+    },
+
+    onKeydown: function(event) {
+      // FIXME want to define these somewhere, core-a11y-keys?
```

Comment:
Should core-a11y-keys be an element, given what we know about the performance problems with using elements for non-visual behavior? If not, should it be some kind of mixin / behavior?